pymongo/helpers.py

```diff
@@ -134,13 +156,19 @@ def _index_document(index_list):
 
 
 def _check_command_response(
-    response, max_wire_version, allowable_errors=None, parse_write_concern_error=False
-):
+    response: Dict[str, Any],
```

Comment:
We should try to use `Mapping` instead of `Dict` where applicable, and use `dict` where not applicable as `Dict` is deprecated as of 3.9: https://docs.python.org/3/library/typing.html#typing.Dict.